#!/usr/bin/env python3

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "dv_ros_capture"
gen = ParameterGenerator()

gen.add("global_hold", bool_t, 0, "Enable automatic exposure", False)

bias_sensitivity_enum = gen.enum([gen.const("VeryLow", int_t, 0, "Very low bias sensitivity"),
                                  gen.const("Low", int_t, 1, "Low bias sensitivity"),
                                  gen.const("Default", int_t, 2, "Default bias sensitivity"),
                                  gen.const("High", int_t, 3, "High bias sensitivity"),
                                  gen.const("VeryHigh", int_t, 4, "Very high bias sensitivity")],
                                 "DVS bias sensitivity preset")
gen.add("bias_sensitivity", int_t, 0,
        "Set DVS bias sensitivity preset.", 2, 0, 4,
        edit_method=bias_sensitivity_enum)

gen.add("noise_filtering", bool_t, 0, "Enable noise filtering on event stream", False)
gen.add("noise_background_activity_time", int_t, 0,
        "Event noise filter background activity time threshold in microseconds", 2000, 1, 10000)

exit(gen.generate(PACKAGE, "capture_node", "DVXplorer"))
